src/components/Molecules/Task/index.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+import CheckBox from "../../Atoms/Checkbox/index";
+import Input from "../../Atoms/Input/index";
+import EditButton from "../../Atoms/EditButton/index";
+import COLOR from "../../../variables/color";
+import TEXT from "../../../variables/texts";
+
+const Task = ({
+  defalutValue,
+  defalutIsEditing,
+  onEditComplete,
+  onTaskComplete,
+  taskName,
+}) => {
+  // タスクが編集中かどうかの状態管理
+  const [isEditing, setIsEditing] = useState({ defalutIsEditing });
+
+  onTaskComplete = () => {
+    setIsEditing(false);
+    console.log("task completed");
+  };
+
+  const onEditStart = () => {
+    setIsEditing(true);
+  };
+
+  onEditComplete = (taskName) => {
+    setIsEditing(false);
+    console.log(`taskname changed: ${taskName}`);
+    console.log({ taskName });
+  };

```

Comment:
```suggestion
  const handleCompleteEdit = (taskName) => {
    setIsEditing(false);
    onEditComplete(TaskName);
  };
```
こちらもこんな感じに修正するのがいいかと！